app/src/main/java/com/raulh82vlc/functionalkotlinarrow/data/network/NetworkDataSource.kt

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (C) 2018 Raul Hernandez Lopez
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.raulh82vlc.functionalkotlinarrow.data.network
+
+import arrow.Kind
+import arrow.core.Try
+import arrow.core.right
+import arrow.effects.IO
+import arrow.effects.async
+import arrow.effects.fix
+import arrow.effects.monadError
+import arrow.effects.typeclasses.Async
+import arrow.typeclasses.binding
+import com.google.gson.Gson
+import com.jakewharton.retrofit2.adapter.kotlin.coroutines.experimental.CoroutineCallAdapterFactory
```

Comment:
I was trying here to get a Deferred and play with it before heading to a normal _Call_ object